src/commands/synthetics/run-test.ts

```diff
@@ -177,29 +179,60 @@ export class RunTestCommand extends Command {
   }
 
   private async getTestsList(api: APIHelper) {
-    if (this.testSearchQuery) {
-      const testSearchResults = await api.searchTests(this.testSearchQuery)
+    if (this.config.testSearchQuery) {
+      const testSearchResults = await api.searchTests(this.config.testSearchQuery)
 
       return testSearchResults.tests.map((test) => ({config: this.config.global, id: test.public_id}))
     }
 
-    const listOfGlobs = this.fileGlobs || [this.config.files]
-
-    const suites = (await Promise.all(listOfGlobs.map((glob: string) => getSuites(glob, this.reporter!))))
+    const suites = (await Promise.all(this.config.fileGlobs.map((glob: string) => getSuites(glob, this.reporter!))))
       .reduce((acc, val) => acc.concat(val), [])
       .map((suite) => suite.tests)
       .filter((suiteTests) => !!suiteTests)
 
     const testsToTrigger = suites
       .reduce((acc, suiteTests) => acc.concat(suiteTests), [])
       .map((test) => ({
-        config: {...this.config!.global, ...test.config},
+        config: {...this.config.global, ...test.config},
         id: test.id,
       }))
 
     return testsToTrigger
   }
 
+  private async resolveConfig() {
+    // Default < ENV < file < CLI
```

Comment:
Let's have environment defined config supersede file-defined config
eg. `Default < file < ENV < CLI`

Reply:
I am not sure about this one. The file is parsed after so it may change the override for customers.

Reply:
+1  that ENV should overwrite the file.
If we did the other way around before, we should bump a minor and notify the breaking change in the release.